hstream-store/HStream/Store/Internal/LogDevice.hs

```diff
@@ -127,6 +129,17 @@ trim client logid lsn =
                            (c_ld_client_trim client' logid lsn)
                            (E.throwSubmitIfNotOK callStack)
 
+findTime' :: HasCallStack => LDClient -> C_LogID -> C_Timestamp -> IO LSN
+findTime' client logid ts = findTime client logid ts c_accuracy_strict
+
+findTime :: HasCallStack => LDClient -> C_LogID -> C_Timestamp -> C_ACCURACY -> IO LSN
+findTime client logid ts accuracy =
+  withForeignPtr client $ \client' -> do
+    (errno, lsn, _) <- withAsyncPrimUnsafe2' (0 :: ErrorCode) LSN_INVALID
+      (c_ld_client_find_time client' logid ts accuracy) E.throwSubmitIfNotOK'
+    void $ E.throwStreamErrorIfNotOK' errno
+    return lsn
+
```

Comment:
the no `'` version is the function you want user to use.  Also,  a sum type of `C_ACCURACY` is better than CInt, e.g.

```
data FindKeyAccuracy 
  = FindKeyStrict
  | FindKeyApproximate
```